### ISSUE TITLE
acceptsInputs should also receive the foreign node

### ISSUE THREAD

**Issue by smeijer** [slothking-online/diagram#76]
When I'm dragging a link from `node A` to `node B`, the `acceptInputs` of `node B` is being triggered, validating if the link is allowed.

The signature of `acceptInputs` is `definition`, `definitions`. The first one, holds information about the `target`, `node B`, the last one contains information about all `definitions`.

Information about the `source`, `node A` is missing here. We could definitely use that. Perhaps `node A` is only allowed to have 2 outgoing links. Or `node B` is not allowed to have links from `node A` for some reason. Perhaps even based on data we have saved before in `node A`.

Basically, I need to be able to decide if the link is allowed, based on the input and output node.

**Comment by aexol** [comment]
Ok I will implement this too
